### PR TITLE
Provide option to log as Json

### PR DESCRIPTION
**Pull request by JoergSiebahn**

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -109,6 +109,17 @@ be created in the `admin` database when AWS DocumentDB is used.
...
+#### Logging
+
+* `ENABLE_JSON_LOGGING` _boolean
+  * Enables logging as Json if `true` (case-insensitive).
+    Configuration errors are never logged as Json.
+    Each log will be one line of Json, containing the keys:
+    * `level`: `"INFO"`, `"WARN"` or `"ERROR"`.
+    * `message`: The log message.
+    * `exception`: The exception, if any.
+    * `mdc`: Object with additional key-value information, if any.
```
Comment (christopher-cudennec) [inline docs/deployment.md:121]: Information about the default value would be helpful.

File: src/main/java/com/sdase/k8s/operator/mongodb/controller/MongoDbController.java [hunk 2]
```diff
@@ -43,61 +46,72 @@ public MongoDbController(
...
-        if (!databaseDeleted) {
-          if (context.getRetryInfo().map(RetryInfo::isLastAttempt).orElse(false)) {
-            LOG.warn(
-                "Last attempt to delete database {} failed. Skipping.",
-                deleteDatabaseTask.databaseName());
-            return DeleteControl.defaultDelete();
+    try (var ignoredMdcNamespace =
+            MDC.putCloseable(MDC_RESOURCE_NAMESPACE_KEY, resource.getMetadata().getNamespace());
+        var ignoredMdcName =
+            MDC.putCloseable(MDC_RESOURCE_NAME_KEY, resource.getMetadata().getName())) {
```
Comment (christopher-cudennec) [inline src/main/java/com/sdase/k8s/operator/mongodb/controller/MongoDbController.java:52]: `resource.getMetadata().getNamespace()` + `...getName()` is called multiple times in `cleanup` and `reconcile`. I'd rather extract the values as variables at the beginning of the methods.

File: src/main/java/com/sdase/k8s/operator/mongodb/logging/LogConfigurer.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.contrib.jackson.JacksonJsonFormatter;
+import ch.qos.logback.contrib.json.classic.JsonLayout;
+import ch.qos.logback.core.Appender;
+import ch.qos.logback.core.ConsoleAppender;
+import ch.qos.logback.core.encoder.LayoutWrappingEncoder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogConfigurer {
```
Comment (christopher-cudennec) [inline src/main/java/com/sdase/k8s/operator/mongodb/logging/LogConfigurer.java:14]: Was this code copied 1:1 from sda-dw-commons or did you make any changes?

File: docs/deployment.md (hunk 1 above)
Comment (JoergSiebahn) [reply docs/deployment.md]: fixed

File: src/main/java/com/sdase/k8s/operator/mongodb/controller/MongoDbController.java (hunk 2 above)
Comment (JoergSiebahn) [reply src/main/java/com/sdase/k8s/operator/mongodb/controller/MongoDbController.java]: Changed … though I don't think, that it increases readability a lot. Even Sonar did not complain.

**Review by christopher-cudennec — APPROVED** [review]

**Pull request merged**